internal/provider/clickhouse_cluster_resource.go

```diff
@@ -493,6 +516,12 @@ func (r *ClickhouseClusterResource) Schema(ctx context.Context, req resource.Sch
 			"access": AccessSchemaBlock(),
 			"config": clickhouseConfigSchemaBlock(),
 			// maintenance window
+			"custom_certificate": schema.SingleNestedBlock{
+				Attributes:          clickhouseCustomCertificateSchema(),
+				PlanModifiers:       []planmodifier.Object{objectplanmodifier.UseStateForUnknown()},
+				MarkdownDescription: "Custom TLS certificate",
+				Validators:          []validator.Object{&clickhouseCustomCertificateValidator{}},
```

Comment:
As I see it just checks that all 3 params are set or unset, you can use stringvalidator.AlsoRequires as validator for every field

Reply:
No, `root_ca` is optional (if customer uses certificate from some public vendor)